### PR TITLE
Add Jetpack Plugin Review Prompt to Scan Flow

### PR DESCRIPTION
**Pull request by elliottprogrammer**
#### Changes proposed in this Pull Request

* Adds a review prompt to the main Scan page, only if the user has completed a successful threat auto-fix.

Screenshot on 2021-03-25 at 17-47-56

#### Testing instructions

1. Checkout and run this PR ('yarn start`)
2. Select (or create with JN) a jetpack site with Scan enabled. 
3. Go to  'http://calypso.localhost:3000/scan/:site` 
4. Open dev tools and start analytics logging by running `localStorage.setItem('debug', 'calypso:analytics');` via the console.
5. Go to the `/wp-content/plugins/` directory and modify any main plugin file and add the line:
    - `$eicar = "X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*";`
    - ( this will trigger a scan threat that Jetpack can auto-fix in the next steps)
6. Click "Scan now" button to start a scan.
7. When scan finishes (you should have 1 scan threat), click "Auto fix 1 threat", and confirm "Fix all threats" button. 
8. When auto-fix finishes, you should see the review prompt at the bottom of the Scan card like shown in the image above.
9. Verify the `calypso_jetpack_scan_review_prompt_view' Tracks event logged in the console.
10. Refresh the page and verify that the review prompt is persistent.
11. Dismiss the prompt by pressing the X in the top right corner.
12. Verify that the `calypso_jetpack_scan_review_prompt_dismiss` event is logged.
13. Unset the `jetpack-review-prompt` preference
   * Screen Shot 2021-03-24 at 12 22 33 PM
 14. Repeat steps 5 - 8 again.
 15. Click the "Leave a Review" button
 16. Check that the `calypso_jetpack_scan_review_prompt_dismiss` event logged with `reviewed: true`.
 17. Verify the prompt is no longer rendered and the that it opens the Jetpack plugin review page.

Related to # 1192874419362946-as-1199999287485857

### CODE REVIEW

File: client/my-sites/scan/threat-fixed-review-prompt/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import './style.scss';
+
+const ThreatFixedReviewPrompt: React.FC = () => {
+	const translate = useTranslate();
+	const moment = useLocalizedMoment();
+	const dispatch = useDispatch();
+
+	const isReviewPromptValid = useSelector( ( state ) => getIsValid( state, 'scan' ) );
+	const isReviewPromptDismissed = useSelector( ( state ) => getIsDismissed( state, 'scan' ) );
+	const lastThreatFixedDate = useSelector(
```
Comment (cbauerman) [inline client/my-sites/scan/threat-fixed-review-prompt/index.tsx:36]: whoops! we could definitely at a `getValidFrom` selector to existing selector files, but this works great 👍

**Review by cbauerman — APPROVED** [review]
This looks great! LGTM.

Screen Shot 2021-03-26 at 11 50 03 AM
To my eye this looks like it should have an equal margins on the top and left, but no reason not to merge!

**Comment by cbauerman** [comment]
> To my eye this looks like it should have an equal margins on the top and left, but no reason not to merge!


"top and right", that is 🤦

**Comment by elliottprogrammer** [comment]
Closing this in favor of #51607 instead.

**Pull request closed without merging**